tsl/src/bgw_policy/job.c

```diff
@@ -840,5 +860,9 @@ job_execute(BgwJob *job)
 	if (transaction_started)
 		CommitTransactionCommand();
 
+	/* Drop portal if it was created */
+	if (portal_created)
+		PortalDrop(portal, false);
```

Comment:
Leaving ActivePortal pointing to pfree'd memory doesn't strike me as a great idea.

Reply:
Another question is what will happen if an exception will be thrown between creating a portal and this line.

Reply:
> Leaving ActivePortal pointing to pfree'd memory doesn't strike me as a great idea.

You're correct... fixed!

Reply:
> Another question is what will happen if an exception will be thrown between creating a portal and this line.

This is responsibility of `PostgresMain` by calling `PortalErrorCleanup` no? https://github.com/postgres/postgres/blob/master/src/backend/tcop/postgres.c#L4245

Reply:
If you are sure that job_execute() is never called from any PG_TRY's then probably that's the case.